envoy-control-runner/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/debug/SnapshotDebugController.kt

```diff
@@ -90,6 +94,50 @@ class SnapshotDebugController(controlPlane: ControlPlane) {
         }
     }
 
+    @GetMapping("/snapshot-global/{service}")
+    fun serviceSnapshot(
+        @PathVariable service: String,
+        @RequestParam dc: String?,
+        @RequestParam(defaultValue = "false") xds: Boolean
+    ): ResponseEntity<EndpointInfoList> {
+        val updateResult = snapshotUpdater.getGlobalSnapshot()
+
+        val globalSnapshot = if (xds) {
+            updateResult?.xdsSnapshot
+        } else {
+            updateResult?.adsSnapshot
+        }
+
+        val endpoints = extractEndpoints(globalSnapshot, service)
+
+        if (endpoints == null) {
+            return ResponseEntity(HttpStatus.NOT_FOUND)
+        }
+
+        val clusterInfos = endpoints.endpointsList
```

Comment:
I would extract logic to a method named like: 
```suggestion
        val clusterInfos = getClusterInformation(endpoints, dc)
```